mne/datasets/limo/limo.py

```diff
@@ -0,0 +1,253 @@
+        Local path to the given data file. This path is contained inside a list
+        of length one, for compatibility.
+
+    Notes
+    -----
+    For example, one could do:
+
+        >>> from mne.datasets import limo
+        >>> url = 'http://datashare.is.ed.ac.uk/download/DS_10283_2189.zip'
+        >>> limo.data_path(url, os.getenv('HOME') + '/datasets') # doctest:+SKIP
+
+    This would download the LIMO data file to the 'datasets' folder,
+    and prompt the user to save the 'datasets' path to the mne-python config,
+    if it isn't there already.
+
+    References
+    ----------           
+    .. [1] Guillaume, Rousselet. (2016). LIMO EEG Dataset, [dataset].
+           University of Edinburgh, Centre for Clinical Brain Sciences.
+           https://doi.org/10.7488/ds/1556.
+    .. [2] https://datashare.is.ed.ac.uk/handle/10283/2189?show=full
+    """  # noqa: E501
+    # set destination path for download
+    key = 'MNE_DATASETS_LIMO_PATH'
+    name = 'LIMO'
+    path = _get_path(path, key, name)
+    destination = re.sub('/download', '', _url_to_local_path(url, op.join(path, 'MNE-limo-data')))
+    limo_dir = op.split(destination)[0]
+
+    # check if zip.-folders are contained
+    is_zip = True if '.zip' in [x.lower() for x in op.splitext(destination)] else False
+
+    # fetch the file from url
+    do_unzip = False
+    if not op.isfile(destination) or force_update:
+        if op.isfile(destination):
+            os.remove(destination)
+        if not op.isdir(op.dirname(destination)):
+            os.makedirs(op.dirname(destination))
+        _fetch_file(url, destination, print_destination=False)
+        do_unzip = True
+
+    # decompress if necessary
+    if is_zip and do_unzip:
+        z1 = zipfile.ZipFile(destination)
+        files = z1.namelist()
+        if do_unzip:
+            stdout.write('Decompressing %g files from\n'
+                         '"%s" ...' % (len(files), destination))
+            z1.extractall(limo_dir)
+            stdout.write(' [done]\n')
+        z1.close()
+        # list files in .zip-folder and look for further .zip-folders to decompress
+        extracted_files = [op.join(limo_dir, f) for f in files]
+        sub_files = [e for e in extracted_files if ".zip" in e or ".tar" in e]
+
+        if sub_files:
+            stdout.write('Decompressing the extracted files ...\n')
+            for i in sub_files:
+                if '.zip' in i:
+                    sub_dir = op.splitext(i)[0]
+                    z2 = zipfile.ZipFile(i)
+                    z2.extractall(limo_dir)
+                    z2.close()
+                    # continue decompressing if necessary
+                    sub_sub_files = [j for j in os.listdir(sub_dir) if '.zip' in j]
+                    for k in sub_sub_files:
+                        z3 = zipfile.ZipFile(os.path.join(sub_dir, k))
+                        z3.extractall(sub_dir)
+                        z3.close()
+                elif '.tar' in i:
```

Comment:
Under what conditions is it tar vs. zip?

I think here it shows more clearly why more explicit names are better. What is `i`? :)